Angpysha.LeftTabbedPage/iOS/LeftTabbedPageRenderer.apple.cs

```diff
@@ -1,598 +1,317 @@
+            if (_pageViewController.ViewControllers.Length == 0
+                && LastSelectedIndex >= 0 && LastSelectedIndex < TabbedPage.Children.Count)
             {
-                pageViewController.SetViewControllers(
-                    new[] { ViewControllers[lastSelectedIndex] },
-                   UIPageViewControllerNavigationDirection.Forward,
-                   true, null
-                );
+                _pageViewController.SetViewControllers(new[] {GetViewController(TabbedPage.CurrentPage)},
+                    UIPageViewControllerNavigationDirection.Forward, true, null);
+                
+                //we use Children.IndexOf instead of CurrentPage.TabIndex because TabIndex is 0 here
+                //TODO: why?
+                SetTab(TabbedPage.Children.IndexOf(TabbedPage.CurrentPage));
             }
 
-            // UpdateSwipe(Tabbed.SwipeEnabled);
-            pageViewController.DidFinishAnimating += HandlePageViewControllerDidFinishAnimating;
+            //COMMENTED CODE: looks like useless
+            //_pageViewController.DidFinishAnimating += OnDidFinishAnimating;
         }
 
-
-
-        protected override void Dispose(bool disposing)
+        public override void ViewDidAppear(bool animated)
         {
-            if (disposing)
-            {
-                PageController?.SendDisappearing();
-
-                if (Tabbed != null)
-                {
-                    Tabbed.PropertyChanged -= OnPropertyChanged;
-                    Tabbed.PagesChanged -= OnPagesChanged;
-                    //    TabBar.TabsSelectionChanged -= HandleTabsSelectionChanged;
-                    TabBar.TabSelected -= TabBar_TabSelected;
-                }
-
-                if (pageViewController != null)
-                {
-                    pageViewController.WeakDataSource = null;
-                    pageViewController.DidFinishAnimating -= HandlePageViewControllerDidFinishAnimating;
-                    pageViewController?.Dispose();
-                }
-            }
-
-            base.Dispose(disposing);
+            ((IPageController)Element).SendAppearing();
+            base.ViewDidAppear(animated);
         }
 
-        protected virtual void OnElementChanged(VisualElementChangedEventArgs e)
+        public override void ViewDidLayoutSubviews()
         {
-            ElementChanged?.Invoke(this, e);
-            if (e.NewElement != null)
+            base.ViewDidLayoutSubviews();
+            if (Element == null)
+                return;
+            if (!Element.Bounds.IsEmpty)
+                View.Frame = new RectangleF((float) Element.X, (float) Element.Y, (float) Element.Width,
+                    (float) Element.Height);
+            if (!_queuedSize.IsZero)
             {
-                TabBar.LeftTabbedPageWeak = new WeakReference<Shared.LeftTabbedPage>(leftTabbedPage);
+                Element.Layout(new Rectangle(Element.X, Element.Y, _queuedSize.Width, _queuedSize.Height));
```

Comment:
No, it shouldn't, because Element - is the whole page with tab bar too (you can check the renderer and see, that tab bar is adding to View in which TabPage renders; also you can check the default Xamarin.iOS TabbedRenderer to certain that it shouldn't (https://github.com/xamarin/Xamarin.Forms/blob/80a4dbfb1cd897f478c38d0aafa6a5d67940d2f3/Xamarin.Forms.Platform.iOS/Renderers/TabbedRenderer.cs, line:1454 nothing is subtracted from _queueSize.Height, size fits using ContainerArea property). 

PS. update to fit any page in renderer, not only TabPage was commited